src/todo_collection.js

```diff
@@ -172,12 +165,11 @@ class ToDoCollection {
     this.render();
   }
 
-  delTask(index) {
-    this.todoData.splice(index, 1);
-    for (let i = index; i < this.todoData.length; i += 1) {
-      this.todoData[i].index -= 1;
-    }
-    localStorage.setItem('todoData', JSON.stringify(this.todoData));
+  delTask() {
+    const falseTask = this.todoData.filter((item) => item.completed === false);
+    for (let i = 0; i < falseTask.length; i += 1) falseTask[i].index = i;
+    localStorage.setItem('todoData', JSON.stringify(falseTask));
```

Comment:
- [ ] Kindly as requested by the previous reviewer  ensure that the task object's values for completed are updated upon user actions not only in the local storage but also on the Ui to give a good user experience check the image below 
<img width="1257" alt="Screen Shot 2023-06-09 at 06 35 40" src="https://github.com/lRebornsl/todo-list/assets/60146030/f82dccb9-d479-4da4-93ee-a4d27ce1816f">

Reply:
Solve